apps/src/componentLibrary/tabs/_Tab.tsx

```diff
@@ -149,10 +145,29 @@ const _Tab: React.FunctionComponent<TabsProps> = ({
     </button>
   );
 
+  const preferredTooltip = tooltip || longTooltip;
+
+  useEffect(() => {
+    if (tabTextRef.current) {
+      const textElement = tabTextRef.current;
+      if (textElement.scrollWidth > textElement.clientWidth && !tooltip) {
+        setLongTooltip({
+          tooltipId: 'test123',
+          text: text || '',
+          direction: 'onBottom',
+        });
+      } else {
+        setLongTooltip(undefined);
+      }
+    }
+  }, [text, tooltip]);
```

Comment:
Might also want to recalculate when resizing the screen?

Reply:
Yeah, this is a tricky part.
You can probably follow the pattern we've used in popover for handling resize events though https://github.com/code-dot-org/code-dot-org/commit/ccb49bd43305cf3dd475c8c65e2df8b6d622230a#diff-2aaf9f9a45b7f5245ff3d3ce4aef3494097f267672e163ae64cf945788bcf0b0R95